src/couch_mrview/src/couch_mrview_util.erl

```diff
@@ -800,7 +800,10 @@ reduced_external_size(Tree) ->
     case couch_btree:full_reduce(Tree) of
         {ok, {_, _, Size}} -> Size;
         % return 0 for versions of the reduce function without Size
-        {ok, {_, _}} -> 0
+        {ok, {_, _}} -> 0;
+        % This is for the case when seq_btree or key_byseq_btree is not
+        % null. Reduces will return an integer.
+        {ok, Size} when is_integer(Size) -> Size
```

Comment:
After adding this line, we can get matched clause. However, it looks that the returned size is the number of indexed documents instead of external size for seq_indexed tree and keyseq_indexed tree.

Reply:
Ooops, you're right. I think we may need to modify our seq_tree reduce function:
https://github.com/apache/couchdb/blob/seqt-external-size-bug/src/couch/src/couch_db_updater.erl#L566-L570

I wonder if we can get away with just using couch_btree:size for the seq_indexed trees.
Thoughts @davisp?

Reply:
Depends on what you want the size of. couch_btree:size gives you the number of bytes to store the btree. Depending on how you're storing KV pairs you may have to have a custom reduce to include that info. For instance, in the id_tree we have a custom reduce that includes size information for the document bodies (since they aren't part of the KV structure of the btree) but for the seq_tree there's nothing written beyond what's in the KVs so we don't need a custom reduce.

Reply:
@davisp:  So I think what you're saying is : for seq_btrees and keyseq_btrees, there is no additional data outside the KVs. If external size means uncompressed data that the user is going to store, then the total number of bytes to store seq_btrees is exactly that because we generated it for the user (who specified to include it). Therefore, couch_btree:size should be sufficient. 

Would want to to include the actual reduce value? so something like:
```couch_btree:size(SeqBtree) + ?term_size(reduce_external_size(SeqBtree))```

Reply:
Ah, no, couch_btree:size is compressed on disk size, not external size. If you're doing this for external size calculations you'll have to add that as a custom reduce function. You can take a look at the reduce function on existing views to get an idea of how it'd work:

https://github.com/apache/couchdb/blob/master/src/couch_mrview/src/couch_mrview_util.erl#L1054-L1073